homeassistant/components/todoist/calendar.py

```diff
@@ -107,110 +108,142 @@
 SCAN_INTERVAL = timedelta(minutes=1)
 
 
-def setup_platform(
+def get_system_utc_offset_hours(time_zone: str | None) -> int:
+    """Get the utc offset hours configured for the HA instance.
+
+    This will return 0 if a timezone is not configured.
+    """
+    if time_zone is None:
+        # Default to UTC if no timezone is specified.
+        return 0
+    zone_info = dt.get_time_zone(time_zone)
+    if zone_info is None:
+        # Default to UTC if no timezone specified can not be found.
+        return 0
+
+    utc_offset_delta = zone_info.utcoffset(datetime.utcnow())
+    if utc_offset_delta is None:
+        return 0
+
+    return int(utc_offset_delta.total_seconds() / 60 / 60)
+
+
+async def async_setup_platform(
     hass: HomeAssistant,
     config: ConfigType,
-    add_entities: AddEntitiesCallback,
+    async_add_entities: AddEntitiesCallback,
     discovery_info: DiscoveryInfoType | None = None,
 ) -> None:
     """Set up the Todoist platform."""
-    token = config.get(CONF_TOKEN)
+    token = config[CONF_TOKEN]
+    utc_offset_hours: int = config.get(
+        CONF_UTC_OFFSET_HOURS, get_system_utc_offset_hours(hass.config.time_zone)
+    )
 
     # Look up IDs based on (lowercase) names.
     project_id_lookup = {}
     label_id_lookup = {}
     collaborator_id_lookup = {}
 
-    api = TodoistAPI(token)
-    api.sync()
+    api = TodoistAPIAsync(token)
 
     # Setup devices:
     # Grab all projects.
-    projects = api.state[PROJECTS]
+    projects = await api.get_projects()
+
+    collaborators = []
+    for project in projects:
+        collaborators.extend(await api.get_collaborators(project.id))
```

Comment:
When starting multiple async operations from a loop, it is a common pattern to start then then await at the end with `asyncio.gather` so they can run in parallel

Reply:
Thanks for the tip, I"ll update this and the other comments tomorrow.